### PR TITLE
Implemented Thomas Algorithm in Lua

### PR DESCRIPTION
**Pull request by Vexatos**
I chose not to mutate the input tables. Is that important?

### CODE REVIEW

**Comment by jiegillet** [comment]
It looks good, and not mutating the input is great. We should probably ask someone (paging @Gorzoid ) who knows lua for an actual review though.
 One problem is how you print the system, it doesn't really work since a b and c are diagonal values of the matrix, not rows (and I realized you copied the Julia code, paging @leios ). If you want to prettify it, you can do something like the PHP implementation.

**Comment by Vexatos** [comment]
I just based this one on the Julia implementation.

File: contents/thomas_algorithm/code/lua/thomas.lua [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+local a = {0.0, 2.0, 3.0}
+local b = {1.0, 3.0, 6.0}
+local c = {4.0, 5.0, 0.0}
+local d = {7.0, 5.0, 3.0}
+
+print("The system")
+print(table.unpack(a))
+print(table.unpack(b))
+print(table.unpack(c))
+print(table.unpack(d))
```
Comment (Gorzoid) [inline contents/thomas_algorithm/code/lua/thomas.lua]: As jiegillet said, the system isn't actually this matrix. a,b and c are instead diagonals and d is the "result" column. To properly print the system we would need something like
```lua
print(b[1],	c[1], "",   "",   "|", d[1])
print("",   a[2], b[2], c[2], "|", d[2])
print("",   "",   a[3], b[3], "|", d[3])
```
I don't see much use in making code to properly any size augmented matrix so manually entering the indices should be fine.
It should be fine to use print(table.unpack(... for the solution however.

Comment (Vexatos) [reply contents/thomas_algorithm/code/lua/thomas.lua]: I agree, I based this on the Julia implementation. I will PR a fix to that as well I guess.

**Pull request merged**